docs/developer/soffits/securing_apis.md

```diff
@@ -0,0 +1,94 @@
+that provide separate "front-end" (UI) and "back-end" (server-side) logic are easier to understand,
+maintain, and extend.
+
+In most cases these module-provided APIs follow the principals of <i>Representational State
+Transfer</i> (REST) and use JSON as a data exchange format.  Both of these practices are also
+preferred and recommended.
+
+Often these APIs require some information from a trusted source.  For example, they may need to know
+the identity and/or characteristics of the user, may provide access to sensitive data, or both.
+
+uPortal provides a simple, secure way of sharing this information with APIs.  This capability is
+based on the `SoffitApiPreAuthenticatedProcessingFilter` and [Spring Security][].  The Java class
+files for this feature are available in the `uPortal-soffit-renderer` dependency.
+
+:notebook:  It is not necessary to be a Soffit to use these features.
+
+## Example Implementation
+
+This capability is based on the `SoffitApiPreAuthenticatedProcessingFilter` and [Spring Security][].
+The Java class files for this feature are available in the `uPortal-soffit-renderer` dependency.
+
+The `SoffitApiPreAuthenticatedProcessingFilter` relies on a `Bearer` token in the `Authorization`
+HTTP header.  You can obtain an appropriate token from uPortal's `userinfo` endpoint, which is found
+at the following URI: `/uPortal/api/v5-1/userinfo`.  Any portal user (including unauthenticated
+users) may obtain a personal token from this endpoint.  This token should be obtained with
+JavaScript and passed (also with JavaScript) as a `Bearer` token in the `Authorization` header in
+API requests.
+
+Refer to the sections below to secure API endpoints and access information about the portal user.
+
+### `uPortal-soffit-renderer` Dependency (Gradle Example)
+
+```groovy
+compile "org.jasig.portal:uPortal-soffit-renderer:${uPortalVersion}"
+```
+
+### Spring Security Configuration
+
+This example defines the following rules for accessing API enpoints that begin with `/api`:
+
+  - Authenticated users may send `GET` and `POST` requests
+  - No users may send any other type of request
+
+```java
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .addFilter(preAuthenticatedProcessingFilter())
+            .authorizeRequests()
+                .antMatchers(HttpMethod.GET,"/api/**").authenticated()
+                .antMatchers(HttpMethod.POST,"/api/**").authenticated()
+                .antMatchers(HttpMethod.DELETE,"/api/**").denyAll()
+                .antMatchers(HttpMethod.PUT,"/api/**").denyAll()
+                .anyRequest().permitAll();
+    }
+
+    @Bean
+    public AuthenticationManager authenticationManager() {
+        return new SoffitApiAuthenticationManager();
+    }
```

Comment:
I'm pretty confident this configuration step isn't necessary.  The `SoffitApiAuthenticationManager` doesn't need to be a Spring-managed bean.  We can remove this method -- as well as the line `rslt.setAuthenticationManager(authenticationManager());` (below) -- in favor of calling `setAuthenticationManager(new SoffitApiAuthenticationManager());` in the constructor of  `SoffitApiPreAuthenticatedProcessingFilter `.

Reply:
**UPDATE:**  I tried to remove/simplify this step and had no luck at all.  This variant works;  others don't seem to.